### DISCUSSION TITLE
Old Cypress vs new Next.js doesn't work

### DISCUSSION THREAD

**Discussion by hyperscientist** [DevSkillsHQ/discusssions#28] [Help] [answered]
Hi, I'm currently working on a test and my app based on Next.js 13 with App Directory stack doesn't work well with Cypress 9.7-based test project. If I'm running the tests locally in a headful mode (`npx cypress open`) then they run successfully or if I update Cypress locally to the latest version then tests work perfectly in either headful or headless mode, but sadly if I push a project with updated Cypress then GitHub workflow will fail as it is not compatible with latest Cypress.

Do you may have any solution or could maybe improve gh workflow in the next few days or should I work around it?

**Comment by fenske** [comment]
Hi, @fullturbodev. I'm Anton from Alva (the coding test provider). Just briefly looked into your submission and here is what I've found so far:
- I've managed to run Cypress tests successfully using the provided setup with Cypress 9.7 in a fresh environment (you can try it yourself by navigating to https://gitpod.io/new/#https://github.com/ffc1e12/multi-step-form-level-1_a326160-88dd92).
- Something weird indeed happens in the pipeline (it seems like the test runner gets stuck at some point. you can check the artifacts from [this build](https://github.com/ffc1e12/multi-step-form-level-1_a326160-88dd92/actions/runs/5068646122)).

I saw you were still working on your task. When you're done, could you ping me in this thread? If we haven't figured it out by then, I'll help you address this in an alternative way.

**Comment by fenske** [comment, marked as answer]
@fullturbodev I've had a closer look, and there is an issue with the NextJS router that wouldn't work in the pipeline only. 

Left more details for the hiring team [here](https://github.com/ffc1e12/multi-step-form-level-1_a326160-88dd92/pull/1#issuecomment-1564929954). 

So feel free to ignore this problem, as for them it'll look like your tests have passed.

I wish you the best of luck with the rest of your interview process!

BR,
Anton.
